RecentActivity/src/org/sleuthkit/autopsy/recentactivity/ExtractEdge.java

```diff
@@ -0,0 +1,197 @@
+import org.sleuthkit.datamodel.TskCoreException;
+
+public class ExtractEdge extends Extract{
+    
+    private static final Logger logger = Logger.getLogger(ExtractIE.class.getName());
+    private final IngestServices services = IngestServices.getInstance();
+    private final String moduleTempResultsDir;
+    private Content dataSource;
+    private IngestJobContext context;
+    
+    private static String ESE_TOOL_NAME = "ESEDatabaseView.exe";
+    private static File ESE_TOOL_FILE;
+    private static String EDGE_WEBCACHE_NAME = "WebCacheV01.dat";
+    private static String EDGE = "Edge";
+
+    ExtractEdge() throws NoCurrentCaseException{
+        moduleName = NbBundle.getMessage(Chrome.class, "ExtractEdge.moduleName");
+        moduleTempResultsDir = RAImageIngestModule.getRATempPath(Case.getCurrentCaseThrows(), EDGE) + File.separator + "results";
+    }
+    
+    @Override
+    void process(Content dataSource, IngestJobContext context) {
+        this.dataSource = dataSource;
+        this.context = context;
+        dataFound = false;
+        
+        this.processWebCache();
+        
+         // Bookmarks come from spartan.edb different file
+        this.getBookmark(); // Not implemented yet
+    }
+    
+    void processWebCache(){
+        Path path = Paths.get("ESEDatabaseView", ESE_TOOL_NAME);
+        ESE_TOOL_FILE =  InstalledFileLocator.getDefault().locate(path.toString(), ExtractEdge.class.getPackage().getName(), false); //NON-NLS
+        if (ESE_TOOL_FILE == null) {
+            this.addErrorMessage(
+                    NbBundle.getMessage(this.getClass(), "ExtractEdge.process.errMsg.unableFindESEViewer", this.getName()));
+            logger.log(Level.SEVERE, "Error finding ESEDatabaseViewer program "); //NON-NLS
+        }
+        
+         final String esedumper = ESE_TOOL_FILE.getAbsolutePath();
+
+        // get WebCacheV01.dat files
+        org.sleuthkit.autopsy.casemodule.services.FileManager fileManager = currentCase.getServices().getFileManager();
+        List<AbstractFile> webCachFiles;
+        try {
+            webCachFiles = fileManager.findFiles(dataSource, EDGE_WEBCACHE_NAME); //NON-NLS
+        } catch (TskCoreException ex) {
+            this.addErrorMessage(NbBundle.getMessage(this.getClass(), "ExtractEdge.process.errMsg.errGettingWebCacheFiles",
+                    this.getName()));
+            logger.log(Level.WARNING, "Error fetching 'index.data' files for Internet Explorer history."); //NON-NLS
+            return;
+        }
+
+        if (webCachFiles.isEmpty()) {
+            String msg = NbBundle.getMessage(this.getClass(), "ExtractEdge.process.errMsg.noWebCachFiles");
+            logger.log(Level.INFO, msg);
+            return;
+        }
+
+        dataFound = true;   
+        
+        if(!PlatformUtil.isWindowsOS()){
+            logger.log(Level.WARNING, "Edge data found, unable to parse on non-windows system."); //NON-NLS
+            return;
+        }
```

Comment:
It seems like this should happen up front somewhere, perhaps as the very first line of the calling process() method.

Reply:
Brian's requirement in the Jira was that we log that there are Edge files, but we cannot process them.
I can put the check in the beginning of process, but we need to continue far enough to determine if there are edge files.  I honestly didn't know if what I did here would be enough to fufill the requirement and now that I type this I realize that it might not be.  I am not sure the effect of the dataFound boolean on the gui, if any.

Reply:
How about doing the query for the files at the start of the process() method? You can then do an early exit if you get no files, can get the presence of the files logged still do an early exit if the platform is not Windows and after that an early exit if the executable cannot be located (assuming that the same executable will be used by all of the  sub-methods; the location could be a local variable passed as an argument to each of the processXXX() methods by the process() method).

Reply:
I moved it.  You'll have to let me know what you think of the flow.

Reply:
The flow looks good!